Started writing instructions for Settings

jaceybee opened the pull request:
Wrote instructions for app settings, noted places where I need guidance

jaceybee committed:
Started writing instructions for Settings

Wrote instructions for app settings, noted places where I need guidance

brotskydotcom2 commented on docs/instructions.md:
```diff
@@ -75,10 +74,44 @@ After following the steps in [Quick Start](#quick-start-app) above:
+
+**Allow Whisper to Access** has four sections:
+
+* **Bluetooth** — allowing Whisper access to Bluetooth is what lets you start and join in-person conversations automatically when you tap Whisper or Listen. We highly recommend allowing Bluetooth access!
```
Actually it's required to use the app at all.  If you don't turn on Bluetooth access the app will just show a link saying "Turn on Bluetooth to continue..." which will open the Settings app.  (You can see this by turning off Bluetooth access.)

jaceybee replied:
Okay, I'll document this as it currently is; see issue #24 for future ideas

brotskydotcom2 commented on docs/instructions.md:
```diff
@@ -75,10 +74,44 @@ After following the steps in [Quick Start](#quick-start-app) above:
+
+* **Bluetooth** — allowing Whisper access to Bluetooth is what lets you start and join in-person conversations automatically when you tap Whisper or Listen. We highly recommend allowing Bluetooth access!
+* **Siri & Search** — tapping this option brings you to a menu where you can refine how much access Apple has to how you use Whisper, as well as whether you can use the built-in voice command functions to find and open Whisper
+* **Notifications** — tapping this option brings you to a menu where you can change what notifications (if any) the Whisper app will give you, the style of those notifications, and whether you'll see whispers on your lock screen
```
We recommend that Listeners allow notifications because otherwise, when the app is in the background, Listeners will miss that the Whisperer is trying to get their attention (by playing a sound).

jaceybee replied:
Will just add slight rewording of this to the existing line

brotskydotcom2 commented on docs/instructions.md:
```diff
@@ -75,10 +74,44 @@ After following the steps in [Quick Start](#quick-start-app) above:
+* **Bluetooth** — allowing Whisper access to Bluetooth is what lets you start and join in-person conversations automatically when you tap Whisper or Listen. We highly recommend allowing Bluetooth access!
+* **Siri & Search** — tapping this option brings you to a menu where you can refine how much access Apple has to how you use Whisper, as well as whether you can use the built-in voice command functions to find and open Whisper
+* **Notifications** — tapping this option brings you to a menu where you can change what notifications (if any) the Whisper app will give you, the style of those notifications, and whether you'll see whispers on your lock screen
+  * This is where you choose whether the app alerts you with a sound when a new whisper comes in; to choose a particular alert sound, go to "Alert Sound" in Whisper Settings
```
Actually, it's the Whisperer who chooses the alert sound.  The "Alert Sound" setting is only used when Whispering (to set the initial sound).  Now that Whisperers can select this sound in the Whisper screen, we might want to remove the "Alert Sound" setting completely because it's confusing for Listeners to have one that does nothing :).  Let me know what you think.

jaceybee replied:
I think it makes the most sense to remove this pref completely, and have Whisperers set their alert sound from the Whisperer settings (see issue #25)

jaceybee replied:
Suggested rewrite: "This is where you choose whether the app alerts you with a sound when a new whisper comes in while the app is running in the background. We recommend keeping this on. Note that the Whisperer is the one who chooses what alert sound their whispers make.

brotskydotcom2 commented on docs/instructions.md:
```diff
@@ -75,10 +74,44 @@ After following the steps in [Quick Start](#quick-start-app) above:
+* **Siri & Search** — tapping this option brings you to a menu where you can refine how much access Apple has to how you use Whisper, as well as whether you can use the built-in voice command functions to find and open Whisper
+* **Notifications** — tapping this option brings you to a menu where you can change what notifications (if any) the Whisper app will give you, the style of those notifications, and whether you'll see whispers on your lock screen
+  * This is where you choose whether the app alerts you with a sound when a new whisper comes in; to choose a particular alert sound, go to "Alert Sound" in Whisper Settings
+* **Background App Refresh** — this option is permanently set to on [explanation from Dad]
```
In order to use Bluetooth and internet properly, the app has to be able to send and receive signals while in the background.  This is a permission that the app is granted by Apple and it shows up here not because the user can change it but because Apple's privacy policy is to let you know when apps are operating in the background.

brotskydotcom2 commented on docs/instructions.md:
```diff
@@ -75,10 +74,44 @@ After following the steps in [Quick Start](#quick-start-app) above:
+
+ **Whisper Settings** has five sections:
+
+* **Remember name from last session** — turning this option on means you don't have to type in your name every time you join a conversation in the app. If multiple people use Whisper on the same device, it may make more sense to turn this option off so no one accidentally joins a conversation under the wrong name.
```
Actually "every time you launch the app" - the name is always remembered for as long as the app is running, no matter how many sessions you enter.

brotskydotcom2 commented on docs/instructions.md:
```diff
@@ -75,10 +74,44 @@ After following the steps in [Quick Start](#quick-start-app) above:
+ **Whisper Settings** has five sections:
+
+* **Remember name from last session** — turning this option on means you don't have to type in your name every time you join a conversation in the app. If multiple people use Whisper on the same device, it may make more sense to turn this option off so no one accidentally joins a conversation under the wrong name.
+* **Require Bluetooth listeners to pair** — [[I don't know what this means]]
```
When this is off, the Whisperer will automatically connect to every Bluetooth listener that tries to join the conversation.  Bill was concerned that, once Whisper gets "popular" (ha ha), he might be in a restaurant or some public location and people would snoop on his conversation using Whisper.  So at his request I added this preference.  If it's turned on, then when a listener device wants to pair with a particular whisperer device over bluetooth for the very first time, then both the whisperer and the listener are required to "pair" with each other (same as when you bluetooth pair a headset or speaker to your phone).  SInce we are talking an iOS to iOS pairing, it doesn't happen automatically (the way it does with your speaker): both the whisperer and the listener are shown a pairing screen with a number on it, and each has to confirm they want to pair with the other person who has that number on their screen.  (This is how, for example, phones pair with cars - by code confirmation.). Pairing is a 1-time only thing: once it has been set up between two devices it never has to happen again (at least not until their next major OS upgrade...).

If someone is very security conscious (think Joe Cotsirilos or his mom ;), they turn this setting on to make sure no one can ever eavesdrop on them that they haven't approved.  Bill has never used it, to my knowledge.

jaceybee replied:
This one is going to take some time for me to word properly, just fyi

jaceybee replied:
If I need you to change the pref name, I'll open an issue for it

brotskydotcom2 commented on docs/instructions.md:
```diff
@@ -75,10 +74,44 @@ After following the steps in [Quick Start](#quick-start-app) above:
+
+* **Remember name from last session** — turning this option on means you don't have to type in your name every time you join a conversation in the app. If multiple people use Whisper on the same device, it may make more sense to turn this option off so no one accidentally joins a conversation under the wrong name.
+* **Require Bluetooth listeners to pair** — [[I don't know what this means]]
+* **Read whisper aloud** — turning this option on means your device will read incoming whispers aloud — [[DAD IS THIS OPTION NECESSARY NOW?]]
```
Yes, absolutely, for when people can't look at their device (e.g., Lisa is doing the dishes in the kitchen and Bill is far enough away that she can't hear his device).  Or when I was on the ladder in the kitchen and Bill was in the garage and we were doing connectivity checks on wires and circuit breakers.

jaceybee replied:
I understand what the purpose is, but I think we should take it out and let that be decided in-app. See issue #26

jaceybee replied:
Please prioritize #26 because then I can just remove this line

brotskydotcom2 commented on docs/instructions.md:
```diff
@@ -75,10 +74,44 @@ After following the steps in [Quick Start](#quick-start-app) above:
+* **Remember name from last session** — turning this option on means you don't have to type in your name every time you join a conversation in the app. If multiple people use Whisper on the same device, it may make more sense to turn this option off so no one accidentally joins a conversation under the wrong name.
+* **Require Bluetooth listeners to pair** — [[I don't know what this means]]
+* **Read whisper aloud** — turning this option on means your device will read incoming whispers aloud — [[DAD IS THIS OPTION NECESSARY NOW?]]
+* **Listener sees newest whisper in** — tap on this option to choose whether new whispers appear above (top window) or below (bottom window) earlier whispers from the same conversation
```
You might want to mention that when new whispers are appearing on top, the history is in reverse chronological order, whereas when they appear on the bottom the history is in chronological order.

brotskydotcom2 commented on docs/instructions.md:
```diff
@@ -75,10 +74,44 @@ After following the steps in [Quick Start](#quick-start-app) above:
+* **Require Bluetooth listeners to pair** — [[I don't know what this means]]
+* **Read whisper aloud** — turning this option on means your device will read incoming whispers aloud — [[DAD IS THIS OPTION NECESSARY NOW?]]
+* **Listener sees newest whisper in** — tap on this option to choose whether new whispers appear above (top window) or below (bottom window) earlier whispers from the same conversation
+* **Alert Sound** — tap on this option to choose one of three sounds to alert you to incoming whispers. If you don't want any alert sounds, go into "Notifications" in Allow Whisper to Access and turn off the "Sounds" option
```
See above.  And turning off the sounds option in notifications does NOT stop you from hearing sounds when the app is in the foreground.  It stops you from being NOTIFIED with that sound when the Whisperer has sent it while the app is in the background.

jaceybee replied:
This will go away because of issue #24, please prioritize that one!

brotskydotcom2 replied:
I believe you mean issue #25, not issue #24.

jaceybee replied:
I did!

brotskydotcom2 commented on docs/instructions.md:
```diff
@@ -75,10 +74,44 @@ After following the steps in [Quick Start](#quick-start-app) above:
+* **Listener sees newest whisper in** — tap on this option to choose whether new whispers appear above (top window) or below (bottom window) earlier whispers from the same conversation
+* **Alert Sound** — tap on this option to choose one of three sounds to alert you to incoming whispers. If you don't want any alert sounds, go into "Notifications" in Allow Whisper to Access and turn off the "Sounds" option
+
+#### macOS
```
The macOS settings are actually exactly the same as the iOS settings - it's just that on MacOS the controls for notifications are in a different place than they are on iOS, and the bluetooth permissions are not per-app the way they are in iOS.  So the only difference between iOS and macOS is that top section (controlled by the OS) rather than anything in the bottom section.

brotskydotcom2 reviewed (changes requested):
I've answered your questions in my review comments, and also corrected a few things you thought you knew but actually didn't :).

Go head and keep editing these files, and keep pushing the commits from your machine to GitHub.  Each new commit will get added to this already-open pull request, and once you've completed the instructions and corrections I will take them all at once.

Feel free to add more questions in each commit.  I will review each of them and give you the answers the same way I did here.

jaceybee committed:
Edited Settings instructions

Started altering the way Settings were written following discussion on Zoom/GitHub about Settings content

jaceybee committed:
Merge branch 'clickonetwo:main' into main

jaceybee committed:
Update instructions.md

brotskydotcom2 reviewed (approved):
I'm going to approve with these changes since everything will be changing and I want to clear the pull request.
